src/rdds/lib/resource_usage/resource_usage.py

```diff
@@ -0,0 +1,108 @@
+import resource
+from typing import List
+from os import getpid
+from os.path import join
+
+
+KILOBYTE = float(1E3)
+MEGABYTE = float(1E6)
+GIGABYTE = float(1E9)
+KIBIBYTE = 1024.0
+# Definitions to translate prefixes to Bytes.
+_SI_ISO_SCALE = {'kB': KILOBYTE, 'mB': KILOBYTE**2, 'KB': KIBIBYTE, 'MB': KIBIBYTE**2}
+
+
+
+class ProcessResourceUsage:
+
+    """
+    Interfaces host 'getrusage' tool and proc-fs to acquire process runtime resource usage.
+    Read more in 'man getrusage' and 'man proc'.
+    """
+
+    def __init__(self):
+        self._pid = getpid()
+        self._proc_status: str = join('/proc', '%s/status' % self._pid)
+
+    @staticmethod
+    def get_max_rss() -> float:
+        """
+        On a Linux machine, return the maximum RAM (resident memory) consumed by this process (including threads).
+        :return: Max ram consumed in Bytes
+        """
+        usage_ram_KiB: int = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
+        return float(usage_ram_KiB * KILOBYTE)
+
+    @staticmethod
+    def get_max_rss_children() -> float:
+        """
+        On a Linux machine, return the maximum RAM (resident memory) consumed by largest children (including threads).
+        :return: Max ram consumed in Bytes
+        """
+        usage_ram_KiB: int = resource.getrusage(resource.RUSAGE_CHILDREN).ru_maxrss
+        return float(usage_ram_KiB * KILOBYTE)
+
+    @staticmethod
+    def _get_parse_key_from_proc_file(proc_file_content: str,
+                                      key) -> float:
+        """
+
+        :param proc_file_content: String contents of proc file
+        :param key: Key to fetch
+        :return: Parsed value, as float
+        """
+        i = proc_file_content.index(key)
+        entry: List[str] = proc_file_content[i:].split(None, 3)  # whitespace removal
+        if len(entry) < 3:
+            return 0.0  # invalid format?
```

Comment:
```suggestion
            return 0.0  # invalid format
```
Are you asking or telling the reader about the format?